sdk/storage/Azure.Storage.Blobs/src/PartitionedDownloader.cs

```diff
@@ -141,14 +141,19 @@ internal class PartitionedDownloader
                 // conditions to ensure the blob doesn't change while we're
                 // downloading the remaining segments
                 ETag etag = initialResponse.Value.Details.ETag;
-                BlobRequestConditions conditionsWithEtag = CreateConditionsWithEtag(conditions, etag);
+                BlobRequestConditions conditionsWithEtag = conditions.CreateConditionsWithEtag(etag);
 
                 // Create a queue of tasks that will each download one segment
                 // of the blob.  The queue maintains the order of the segments
                 // so we can keep appending to the end of the destination
                 // stream when each segment finishes.
                 var runningTasks = new Queue<Task<Response<BlobDownloadStreamingResult>>>();
                 runningTasks.Enqueue(initialResponseTask);
+                if (_maxWorkerCount == 1)
```

Comment:
`<= 1`?

Reply:
_maxWorkerCount is  1 or more here due to validation/defaulting. But it won't hurt to have that either.